src/hooks/useDimensions.ts

```diff
@@ -0,0 +1,29 @@
+import { MutableRefObject, useLayoutEffect, useMemo, useState } from "react";
+
+interface DimensionState {
+  width: number;
+  height: number;
+}
+
+export const useDimensions = (ref: MutableRefObject<HTMLElement>) => {
+  const [state, setState] = useState<DimensionState>({ width: 0, height: 0 });
+
+  const observer = useMemo(
+    () =>
+      new (window as any).ResizeObserver((entries) => {
```

Comment:
I'm confused by this type casting why can't you just use `new ResizeObserver...`